assets/src/containers/IndexPage.js

```diff
@@ -21,24 +23,34 @@ const styles = theme => ({
   }
 })
 
-function IndexPage (props) {
+function IndexPage(props) {
   const { classes, match } = props
-  const currentCourseId = match.params.courseId
+  const courseId = match.params.courseId
+  const [loaded, courseInfo] = useCourseInfo(courseId)
```

Comment:
My observation: 
The `useCourseInfo` is calling the `/info` endpoint and I see when the user first type or loads the Index page `/courses/12233`  `/info` is being called from `sideDrawer.js` and `IndexPage.js`. The chrome dev tools it is firing two backend call
![Screen Shot 2019-04-11 at 2 50 31 PM](https://user-images.githubusercontent.com/8579775/55992020-8c878280-5c79-11e9-8610-6f25f3847b3b.png)

Instead, the user just directly goes `/course/1233/grades` bypassing the index page `/info` is fired one time and thereafter he moves to index page the data is pulled from the cache( as part of the `useFetch.js` logic).

Can this `/info` data be stored as a global piece of info somewhere so that subsequent flows would fetch is from that place if not present then make a backend call?

Reply:
Good point. `useFetch` caches responses, but it looks like in this case the requests happen all at the same time, before any responses return. 

The better way would be to call `useCourseInfo` in `App.js` just once, and then pass down the `courseInfo` as props to both `sideDrawer.js` and `IndexPage.js`. I'll make this change.

Reply:
I've implemented this change here: https://github.com/tl-its-umich-edu/my-learning-analytics/pull/481/commits/03f8e0c7cda0508e1ac09bd45015c0b39d591752